frontend/components/puzzle3.jsx

```diff
@@ -1,27 +1,73 @@
 import React from 'react';
-
 import PuzzleHeader from './puzzleHeader';
+import { AnswerAwareDiv } from './puzzle.jsx';
 
 // Answer: Your name
-
+// Answer: [5,4,3,2,1]
 class Puzzle3 extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      combo : ['','','','','']
+    };
+  }
+
+
+  handleClick(input) {
+    var temparray = this.state.combo;
+      var flag = 'full';
+      for (var i = 0; i <= temparray.length; i++) {
+        if ( temparray[i] === '') {
+          temparray.splice(i,1,input);
+          this.setState({ combo: temparray });
```

Comment:
we should update state in a callback since `setState` is actually asynchronous:
```
this.setState(state => {
  const combo = state.combo;
  combo.splice(i, 1, input);
  return { combo };
})
```